backend/app/api/endpoints/rag.py

```
@@ -509,3 +510,68 @@ async def test_retriever_connection(
     except Exception as e:
         raise HTTPException(status_code=500, detail=str(e))
+
+
+@router.get("/index-name", response_model=IndexNameResponse)
+async def get_index_name(
+    knowledge_id: str = Query(..., description="Knowledge base ID"),
+    retriever_name: str = Query(..., description="Retriever name"),
+    retriever_namespace: str = Query(
+        default="default", description="Retriever namespace"
+    ),
+    db: Session = Depends(get_db),
```

Ruff (0.14.10)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)